html/user/edit_email_action.php

```diff
@@ -46,17 +47,18 @@
         // deal with the case where user hasn't set passwd
         // (i.e. passwd is account key)
         //
-        if ($passwd_hash != $user->passwd_hash) {
+        if ($passwd_hash != $user->passwd_hash && !password_verify($passwd_hash,$user->passwd_hash)) {
             $passwd = $user->authenticator;
             $passwd_hash = md5($passwd.$user->email_addr);
         }
-        if ($passwd_hash != $user->passwd_hash) {
+        if ($passwd_hash != $user->passwd_hash && !password_verify($passwd_hash,$user->passwd_hash)) {
             echo tra("Invalid password.");
         } else {
             $passwd_hash = md5($passwd.$email_addr);
+            $database_passwd_hash = password_hash($passwd_hash , PASSWORD_DEFAULT );
             $email_addr = BoincDb::escape_string($email_addr);
             $result = $user->update(
-                "email_addr='$email_addr', passwd_hash='$passwd_hash', email_validated=0"
+                "email_addr='$email_addr', passwd_hash='$database_passwd_hash', email_validated=0"
```

Comment:
[Here](https://github.com/BOINC/boinc/pull/2401/files#diff-2ebf2ee712812c0682531ca5ebf7c713R73) you escape the password hash before its use as part of an SQL clause. If you think it's required, it should be done here as well.

Reply:
After reviewing, the hash does not need to be escaped so I have removed it from the referenced line.